prov/efa/src/efa_domain.c

```diff
@@ -421,7 +421,7 @@ efa_domain_query_mr(struct fid_mr *mr_fid, struct fi_efa_mr_attr *mr_attr)
 	ret = efadv_query_mr(efa_mr->ibv_mr, &attr, sizeof(attr));
 	if (ret) {
 		EFA_WARN(FI_LOG_DOMAIN, "efadv_query_mr failed. err: %d\n", ret);
-		return ret;
+		return (ret == EOPNOTSUPP) ? -FI_EOPNOTSUPP : -FI_EINVAL;
```

Comment:
I think this is true for old kernels that don't have uverbs ioctl support. With most kernels will will get EPROTONOSUPPORT when the driver doesn't have query_mr implemented. I would also avoid pushing it in this PR.